presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java

```diff
@@ -48,17 +47,17 @@
         implements PrestoThriftServiceProvider
 {
     private static final Logger log = Logger.get(RetryingPrestoThriftServiceProvider.class);
-    private final PrestoThriftServiceProvider original;
+    private final ConnectedThriftServiceProvider original;
     private final RetryDriver retry;
 
     @Inject
-    public RetryingPrestoThriftServiceProvider(@NonRetrying PrestoThriftServiceProvider original, @ForRetryDriver ListeningScheduledExecutorService retryExecutor, ThriftConnectorConfig config)
+    public RetryingPrestoThriftServiceProvider(
+            ConnectedThriftServiceProvider original,
+            @ForRetryDriver ListeningScheduledExecutorService retryExecutor,
+            ThriftConnectorConfig config)
     {
         this.original = requireNonNull(original, "original is null");
-        requireNonNull(retryExecutor, "retryExecutor is null");
-        requireNonNull(config, "config is null");
-
-        retry = RetryDriver.retry(retryExecutor)
+        retry = RetryDriver.retry(requireNonNull(retryExecutor, "retryExecutor is null"))
```

Comment:
Why did you remove null check for config? Also, I'd keep retryExecutor null check separately.